sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/apachecommons/lang/Range.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Portions Copyright (c) Microsoft Corporation
+ */
+
+package com.azure.cosmos.implementation.apachecommons.lang;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+public final class Range<T> implements Serializable {
```

Comment:
bug: this is a cosmos class why is this coming here?
@kushagraThapar

Reply:
Yeah, we have our own Range.java class, which doesn't use any of the external libraries other than jackson. 

@anuchandy  can you please confirm if we didn't replace this class by mistake ?

```
package com.azure.cosmos.implementation.routing;

import com.azure.cosmos.BridgeInternal;
import com.azure.cosmos.models.JsonSerializable;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.node.ObjectNode;

import java.io.Serializable;
import java.util.Comparator;

@JsonIgnoreProperties({ "empty", "singleValue", "hashMap" })
public final class Range<T extends Comparable<T>> extends JsonSerializable {
```

Reply:
@moderakh  - The Range.java class that Anu has added should be coming from `org.apache.commons.lang3`
We use commons-lang3 Range.java in `ParallelAsync.java` and `ConsistencyTests2.java`

Reply:
ok correct. looks good thanks.